src/jit/SimdArm64Inl.h

```diff
@@ -140,6 +149,23 @@ static void simdEmitNarrowUnsigned(sljit_compiler* compiler, sljit_s32 rd, sljit
     simdEmitOp(compiler, SimdOp::sqxtun | size | (0x1 << 30), rd, rm, 0);
 }
 
+static void simdEmitShift(sljit_compiler* compiler, sljit_s32 rd, sljit_s32 rn, sljit_s32 rm, int width, SimdOp::ImmSizeType immSize, SimdOp::IntSizeType intSize, bool isRight = false, bool isUnsigned = false)
+{
+    sljit_emit_op2(compiler, SLJIT_AND32, rm, 0, rm, 0, SLJIT_IMM, (1 << width) - 1);
+    uint32_t opcode = SimdOp::dup | immSize | (sljit_get_register_index(SLJIT_GP_REGISTER, rm) << 5) | sljit_get_register_index(SLJIT_SIMD_REG_128, rd);
+    sljit_emit_op_custom(compiler, &opcode, sizeof(uint32_t));
```

Comment:
The jit compiler has a lane replication opcode. Since the code is the same everywhere, it could be extracted into the emitShiftSIMD function.

Reply:
It can't be extracted because the width is different on each call.

Reply:
The first switch(opcode) can also set the width.